backend/msgpack/writer.go

```diff
@@ -219,11 +219,12 @@ func (w *writer) encodeWithLock(
 				break
 			}
 
-			sizeAfter := buffer.Len()
-			if sizeAfter < w.flushSize {
+			// If the buffer isn't big enough continue to the next iteration.
+			if buffer.Len() < w.flushSize {
```

Comment:
super nit: `if sizeAfter := buffer.Len(); sizeAfter < w.flushSize` is slightly easier to comprehend?